core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -202,7 +202,7 @@ private[spark] class PythonRDD(
       this.interrupt()
     }
 
-    override def run(): Unit = Utils.logUncaughtExceptions {
+    override def run(): Unit = Utils.tryLog {
```

Comment:
@rxin, is the concern that a user might write something like

```scala
def foo(): Unit = Utils.tryLog {
   return someCallThatThrowsAnException()
}
```

and thereby skip the exception-handling logic?